api/common/upgradeseries.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2018 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package common
+
+import (
+	"github.com/juju/errors"
+	"gopkg.in/juju/names.v2"
+
+	"github.com/juju/juju/api/base"
+	apiwatcher "github.com/juju/juju/api/watcher"
+	"github.com/juju/juju/apiserver/params"
+	"github.com/juju/juju/watcher"
+)
+
+type UpgradeSeriesAPI struct {
+	facade base.FacadeCaller
+	tag    names.Tag
+}
+
+func NewUpgradeSeriesAPI(facade base.FacadeCaller, tag names.Tag) *UpgradeSeriesAPI {
+	return &UpgradeSeriesAPI{facade: facade, tag: tag}
+}
+
+// WatchActionNotifications returns a StringsWatcher for observing the state of
+// a series upgrade.
+func (u *UpgradeSeriesAPI) WatchUpgradeSeriesNotifications() (watcher.NotifyWatcher, error) {
+	var results params.NotifyWatchResults
+	args := params.Entities{
+		Entities: []params.Entity{{Tag: u.tag.String()}},
+	}
+	err := u.facade.FacadeCall("WatchUpgradeSeriesNotifications", args, &results)
+	if err != nil {
+		return nil, errors.Annotatef(err, "from api/common/upgradeseries FacadeCall(%s)", u.facade.Name())
+	}
+	if len(results.Results) != 1 {
+		return nil, errors.Errorf("expected 1 result, got %d", len(results.Results))
```

Comment:
This error is used very often. An `ag` search an replace to move this into its own function would be great:
```go
somecommonpackage.NewUnexpectedNumResultsError(actualNumResults) 
```